src/main/java/seedu/address/model/Concierge.java

```diff
@@ -165,6 +168,42 @@ public void setRooms(List<Room> rooms) {
         this.rooms.setRooms(rooms);
     }
 
+    /**
+     * Reassigns the booking identified by {@code startDate} in the room identified by {@code roomNumber} to the room
+     * identified by {@code newRoomNumber}.<br>
+     * The following checks are conducted in succession, each possibly throwing an exception:<br>
+     * If the booking does not exist, throw NoBookingFoundException.<br>
+     * If the booking is expired, throw ExpiredBookingException.<br>
+     * If room and new room are equal, throw OriginalRoomReassignException.<br>
+     * If new room has bookings and its first booking is already checked-in, throw ReassignToCheckedInRoomException.<br>
+     * If the booking overlaps with any of new room's bookings, throw OverlappingBookingException.
+     */
+    public void reassignRoom(RoomNumber roomNumber, LocalDate startDate, RoomNumber newRoomNumber) {
+        Room room = rooms.getRoom(roomNumber);
+        Room newRoom = rooms.getRoom(newRoomNumber);
+
+        Booking bookingToReassign = room.getBookings()
+                .getFirstBookingByPredicate(booking -> booking.getBookingPeriod().getStartDate().equals(startDate));
+
+        if (bookingToReassign.isExpired()) {
+            throw new ExpiredBookingException();
+        }
+
+        if (room.equals(newRoom)) {
+            throw new OriginalRoomReassignException();
+        }
+
+        if (newRoom.hasBookings() && newRoom.getBookings().getFirstBooking().getIsCheckedIn()) {
+            throw new ReassignToCheckedInRoomException();
```

Comment:
Is this a subset of `OverlappingBookingException`?

Reply:
Nope, suppose the bookingToReassign is on its last day, and we want to reassign it to another room with a checked-in booking that starts on the same day. The booking would not be considered overlapping, but should not be allowed.

Reply:
I'm curious why this is not allowed o: not very intuitive